derive.h

```diff
@@ -45,28 +49,32 @@ ostream& operator<<(ostream& os, const Expression& ex) {
 
 class CoefExpression : public Expression {
 public:
-	CoefExpression(double a_coef): coef(a_coef) 
+	CoefExpression(double coef): m_coef(coef) 
 	{ }
 	
-	virtual bool is_zero() { return coef == 0.; }
+	virtual bool is_zero() { return coef() == 0.; }
 	virtual shared_ptr<Expression> derivative() const {
 		return std::dynamic_pointer_cast<Expression>(
```

Comment:
Why is dynamic (or any) cast needed here?

Reply:
Indeed it seems that there is no  need.